### DISCUSSION TITLE
memory usage

### DISCUSSION THREAD

**Discussion by euri10** [jesec/rtorrent#11] [Q&A] [answered]
greetings,

I'd rather open a discussion 1st rather than a bug report, hope it's fine.

I recently added to my home server a bunch of monitoring containers, the usual suspects (grafana, cadvisor, prometheus...) and after a few days, I'm wondering if the below looks normal.

This is the memory usage of my containers, on 03/19 at 12:00 approximately I restarted the rtorrent container, after I noticed it was using 23GB or ram.
You can notice that after the restart the memory increase usage over time looks quite linear, I also think he little bump on 03/20 around 20:00 is me downloading a file.

![20210320_0811_1481x311_1616224294](https://user-images.githubusercontent.com/1104190/111862269-8cf42200-8954-11eb-9ab7-fb6ae5799a61.png)

now the container cached memory is the same :+1: 

![20210320_0815_1478x286_1616224537](https://user-images.githubusercontent.com/1104190/111875778-cac86900-899b-11eb-83c0-b4bf21d4ed6b.png)

my rtorrent container runs ~ 1300 torrents, so I wondered if that memory usage made sense

**Comment by jesec** [comment, marked as answer]
I haven't encountered the issue. rTorrent runs perfectly fine for weeks on my old 8GB-RAM server. I seed around 7TB with that machine. 

Note that `cached memory` is very different from `used memory`. 

You should try to monitor the available memory of the entire system.

**Reply by euri10** [reply]
as a reference, how much ram does these 7TB "consume" on your system ?

I seed around 15TB, so I will try to cap the container memory with `--memory` and see if I run out of it

**Reply by jesec** [reply]
RSS: 103116 (approx. 100M)

Again, `cached memory` is very different from `used memory`. 

Cached memory is free memory that has been filled with the contents of blocks on disk. It will be vacated as soon as the space is needed by anything else.

**Reply by euri10** [reply]
mouarf, TIL that displaying stats is way better when you know what they mean exactly !

so I did a `❯ docker stats --no-stream | grep rtorrent`

which tells me I'm using 5.77GB on that container
`82c4dfa9868c   flood_rtorrent_1                0.86%     5.773GiB / 62.94GiB   9.17%     12.3GB / 55.7GB   156GB / 0B        3`

the explanation of the difference is that cadvisor shows it differently (https://stackoverflow.com/questions/55117913/docker-stats-shows-different-data-compared-to-cadvisor)

long story short, seems like everything is fine, I use lot of ram, but most of it is cached, and I wasn't displaying the difference.

**Reply by jesec** [reply]
Good to know. 

BTW: I actually seed approx. 12TB on that machine. I forgot to count a group.
